Marlin/src/HAL/HAL_STM32/persistent_store_impl.cpp

```diff
@@ -24,24 +24,24 @@
 
 #include "../../inc/MarlinConfig.h"
 
-#if ENABLED(EEPROM_SETTINGS)
+#if ENABLED(EEPROM_SETTINGS) && ANY(FLASH_EEPROM_EMULATION, SRAM_EEPROM_EMULATION, SPI_EEPROM, I2C_EEPROM)
```

Comment:
If this doesn't actually need to implement `SPI_EEPROM` or `I2C_EEPROM` then this can be cleaned up a bit more.

Reply:
We don't use it. We can clear SPI_EEPROM, but I2C EEPROM we can keep

Reply:
> If this doesn't actually need to implement `SPI_EEPROM` or `I2C_EEPROM` then this can be cleaned up a bit more.

@thinkyhead SPI_EEPROM can be cleaned , but I2C EEPROM can be saved or not. After all, there are three ways to simulate EEPROM

Reply:
I'm not sure. `SPI_EEPROM` is shared by all platforms, since all platforms can potentially talk to any SPI device.